dbms/programs/compressor/Compressor.cpp

```diff
@@ -85,18 +86,20 @@ int mainEntryClickHouseCompressor(int argc, char ** argv)
         bool use_none = options.count("none");
         unsigned block_size = options["block-size"].as<unsigned>();
 
-        DB::CompressionMethod method = DB::CompressionMethod::LZ4;
+        std::string method_family = "LZ4";
 
         if (use_lz4hc)
-            method = DB::CompressionMethod::LZ4HC;
+            method_family = "LZ4HC";
         else if (use_zstd)
-            method = DB::CompressionMethod::ZSTD;
+            method_family = "ZSTD";
         else if (use_none)
-            method = DB::CompressionMethod::NONE;
+            method_family = "NONE";
 
-        DB::CompressionSettings settings(method, options.count("level")
-            ? options["level"].as<int>()
-            : DB::CompressionSettings::getDefaultLevel(method));
+        std::optional<int> level;
+        if (options.count("level"))
+            level = options["level"].as<int>();
+
+        DB::CompressionCodecPtr codec = DB::CompressionCodecFactory::instance().get(method_family, level);
```

Comment:
You can also introduce `--codec` option that will allow to pass stacked codecs.

Reply:
https://github.com/yandex/ClickHouse/pull/3899/files#diff-238917e94163b6eba9b15c42a9aa2b55